test/unit/src/composite/MultiImplementationCompositeTests.cpp

```diff
@@ -0,0 +1,451 @@
+//
+// Created by Timor Gruber on 11/28/19.
+// Copyright (c) 2019 Takpit. All rights reserved.
+//
+#include <catch2/catch.hpp>
+#include <fff.h>
+
+#include "composite/MultiImplementationCompositeTests.h"
+
+//region Fakes
+
+// Fake declared functions to enable call statistics
+FAKE_VOID_FUNC(compositeActionNoArgsImpl1)
+
+FAKE_VOID_FUNC(compositeActionNoArgsImpl2)
+
+FAKE_VALUE_FUNC(int, compositeFuncNoArgsImpl1)
+
+FAKE_VALUE_FUNC(int, compositeFuncNoArgsImpl2)
+
+FAKE_VOID_FUNC(compositeActionSingleArgImpl1, int)
+
+FAKE_VOID_FUNC(compositeActionSingleArgImpl2, int)
+
+FAKE_VALUE_FUNC(int, compositeFuncSingleArgImpl1, int)
+
+FAKE_VALUE_FUNC(int, compositeFuncSingleArgImpl2, int)
+
+//endregion
+
+//region No Args
+
+DEFINE_IMPLEMENTATION(NoArgsCompositeInterfaceImpl1, compositeActionNoArgs, compositeActionNoArgsImpl1,
+                      compositeFuncNoArgs, compositeFuncNoArgsImpl1)
+
+DEFINE_IMPLEMENTATION(NoArgsCompositeInterfaceImpl2, compositeActionNoArgs, compositeActionNoArgsImpl2,
+                      compositeFuncNoArgs, compositeFuncNoArgsImpl2)
+
+SCENARIO("Multiple implementations of no-args action and func interface "
+         "call different methods")
+{
+    RESET_FAKE(compositeActionNoArgsImpl1)
+    RESET_FAKE(compositeFuncNoArgsImpl1)
+    RESET_FAKE(compositeActionNoArgsImpl2)
+    RESET_FAKE(compositeFuncNoArgsImpl2)
+
+    GIVEN("Implementation 1")
+    {
+        NoArgsCompositeInterface noArgsCompositeInterface{};
+        initNoArgsCompositeInterfaceImpl1(&noArgsCompositeInterface);
+
+        WHEN("Interface action is called")
+        {
+            noArgsCompositeInterface.compositeActionNoArgs();
+
+            THEN("Action implementation 1 is called")
+            {
+                REQUIRE(compositeActionNoArgsImpl1_fake.call_count == 1);
+            }
+
+            THEN("Action implementation 2 is NOT called")
+            {
+                REQUIRE(compositeActionNoArgsImpl2_fake.call_count == 0);
+            }
+            THEN("Func implementation 2 is NOT called")
+            {
+                REQUIRE(compositeFuncNoArgsImpl2_fake.call_count == 0);
+            }
+        }
+        AND_WHEN("Interface func is called")
+        {
+            noArgsCompositeInterface.compositeFuncNoArgs();
+
+            THEN("Func implementation 1 is called")
+            {
+                REQUIRE(compositeFuncNoArgsImpl1_fake.call_count == 1);
+            }
+
+            THEN("Action implementation 2 is NOT called")
+            {
+                REQUIRE(compositeActionNoArgsImpl2_fake.call_count == 0);
+            }
+            THEN("Func implementation 2 is NOT called")
+            {
+                REQUIRE(compositeFuncNoArgsImpl2_fake.call_count == 0);
+            }
```

Comment:
Same for here - Action implementation 2 should (or shouldn't) be checked too? 
I see this triple check only is common for all the next test so I'll not mention it again down below...